src/main/java/net/ccbluex/liquidbounce/utils/login/UserUtils.kt

```diff
@@ -32,64 +45,71 @@ object UserUtils {
+        }
 
-        return response.statusLine.statusCode == 204
+        client.execute(request).use { response ->
+            EntityUtils.consumeQuietly(response.entity)
+            val isValid = response.statusLine.statusCode == 204
+            tokenCache[token] = isValid
+            return isValid
+        }
     }
 
     fun getUsername(uuid: String): String? {
-        val client = HttpClients.createDefault()
-        val request = HttpGet("https://api.mojang.com/user/profiles/$uuid/names")
-        val response = client.execute(request)
-
-        if (response.statusLine.statusCode != 200) {
-            return null
+        uuidCache[uuid]?.let { return it }
+
+        val request = HttpGet("https://api.minecraftservices.com/minecraft/profile/lookup/$uuid")
+
+        client.execute(request).use { response ->
+            if (response.statusLine.statusCode != 200) return null
+            return try {
+                val name = JsonParser().parse(EntityUtils.toString(response.entity))
+                    .asJsonArray
+                    .last()
+                    .asJsonObject["name"]
+                    .asString
+                uuidCache[uuid] = name
+                name
+            } catch (e: Exception) {
+                e.printStackTrace()
+                null
+            }
         }
-
-
-        return JsonParser().parse(EntityUtils.toString(response.entity)).asJsonArray.last().asJsonObject["name"].asString
     }
 
     /**
      * Get UUID of username
      */
-    fun getUUID(username : String) : String {
-        try {
-            // Make a http connection to Mojang API and ask for UUID of username
-            val httpConnection = URL("https://api.mojang.com/users/profiles/minecraft/$username").openConnection() as HttpsURLConnection
-            httpConnection.connectTimeout = 2000
-            httpConnection.readTimeout = 2000
-            httpConnection.requestMethod = "GET"
-            httpConnection.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:25.0) Gecko/20100101 Firefox/25.0")
-            HttpURLConnection.setFollowRedirects(true)
-            httpConnection.doOutput = true
-
-            if (httpConnection.responseCode != 200)
-                return ""
-
-            // Read response content and get id from json
-            InputStreamReader(httpConnection.inputStream).use {
-                val jsonElement = JsonParser().parse(it)
-
+    fun getUUID(username: String): String {
+        usernameCache[username]?.let { return it }
+
+        return try {
+            val url = URL("https://api.minecraftservices.com/minecraft/profile/lookup/name/$username")
+            (url.openConnection() as HttpsURLConnection).apply {
+                connectTimeout = 2000
+                readTimeout = 2000
+                requestMethod = "GET"
+                setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:25.0) Gecko/20100101 Firefox/25.0")
```

Comment:
NT 6.1 :skull:
Windows 7...
Really old, but whatever.

Reply:
I will update it don't worry